framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -158,21 +153,102 @@ private static Pattern anchored(String regex) {
     }
 
     private static boolean isMemberSelect(String s, FlowExpressionContext context) {
-        return parseMemberSelect(s) != null;
+        return parseMemberSelect(s, context) != null;
+    }
+
+    /**
+     * Matches field accesses whose receiver is a method call expression. First of returned pair is
+     * a method call expression and the second is a field or method call expression.
+     *
+     * @param sWithFormalParamNames expression string with replaced occurences of one-based
+     *     parameter index of with formal parameter names
```

Comment:
I don't understand this comment.  That may be because there is no information in the Javadoc for `s` about what it is.
Are there any constrains on `s`?  I think it may have numbered formal parameters, such as `#1.foo()` or `foo(#2)`.  The documentation needs to be specific about the format of strings.

Maybe the two formal parameters should be named `exprWithNamedParameters` and `exprWithNumberedParameters`.